### ISSUE TITLE
no-extraneous-dependencies doesn’t report anything if no dependencies have been specified

### ISSUE THREAD

**Issue by remcohaszing** [benmosher/eslint-plugin-import#1161]
Given the following files:

*package.json*

```json
{
  "name": "foo",
  "version": "1.0.0",
}

```

*.eslintrc.js*

```js
module.exports = {
  plugins: ['import'],
  parserOptions: {
    sourceType: 'module'
  },
  rules: {
    'import/no-extraneous-dependencies': 'error'
  }
}
```

*index.js*

```js
import 'eslint'
```

Install `eslint-plugin-import`, but don’t save it to package.json

```
npm i --no-save eslint eslint-plugin-import
```

When running `eslint .`, no errors are reported, even though it **should** warn that `eslint` needs to be added to *package.json*.

Now add any packages to the dependencies

*package.json*

```json
{
  "name": "foo",
  "version": "1.0.0",
  "dependencies": {
    "foo": "*"
  }
}
```

Now `import/no-extraneous-dependencies` **does** get triggered.

This issue exists either when no dependencies key has been specified in *package.json*, or if it contains an empty object.

**Comment by ljharb** [comment]
hmm, i'm able to reproduce the lack of a warning, but i'm unable to reproduce that it works when you add one. I'll keep looking.
